From 7c701a918b3d08e85a6cfa70a531d019882378dc Mon Sep 17 00:00:00 2001
From: Francois Daoust <fd@tidoust.net>
Date: Sat, 2 Nov 2024 10:59:54 +0100
Subject: [PATCH] Amend syntax of `if()`, drop value of `<if-condition>`

For `if()`, parsing fails on `;?`.

For `<if-condition>`, the problem is that the spec extends the Value Definition
Syntax with a new construct that is not yet supported by the CSS parser.

Both problems have been fixed in CSSTree already, but a new version still needs
to be released. See tracking issue in:
https://github.com/w3c/webref/issues/1378


---
 ed/css/css-values-5.json | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/ed/css/css-values-5.json b/ed/css/css-values-5.json
index 8aa66d177..7b6b54fb7 100644
--- a/ed/css/css-values-5.json
+++ b/ed/css/css-values-5.json
@@ -462,13 +462,12 @@
       "name": "<if()>",
       "href": "https://drafts.csswg.org/css-values-5/#typedef-if",
       "type": "type",
-      "value": "if( [ <if-condition> : <declaration-value>? ; ]* <if-condition> : <declaration-value>? ;? )"
+      "value": "if( [ <if-condition> : <declaration-value>? ; ]* <if-condition> : <declaration-value>? ';'? )"
     },
     {
       "name": "<if-condition>",
       "href": "https://drafts.csswg.org/css-values-5/#typedef-if-condition",
-      "type": "type",
-      "value": "<boolean[ <if-test> ]> | else"
+      "type": "type"
     },
     {
       "name": "<if-test>",
-- 
2.37.1.windows.1

